src/monmove.c

```diff
@@ -141,7 +141,11 @@ register struct monst *mtmp;
 
     if (mtmp->mpeaceful && in_town(u.ux + u.dx, u.uy + u.dy)
         && mtmp->mcansee && m_canseeu(mtmp) && !rn2(3)) {
-        if (Role_if(PM_CONVICT) && !Upolyd) {
+        if (Role_if(PM_CONVICT)
+            && !Upolyd 
+            && (!ublindf || (ublindf
+                             && ublindf->otyp != TOWEL
+                             && ublindf->otyp != BLINDFOLD))) {
```

Comment:
The `ublindf` check in the second part is guaranteed to be true, since if it
weren't, the `!ublindf` part would have been true (thanks to short-circuit
evaluation the second part won't be evaluated if `!ublindf` is true).  So you
could just jump to checking `ublindf->otyp` directly without explicitly
(re)testing that `ublindf` is non-null.

Reply:
Good point! :)